src/bus/policy.c

```diff
@@ -602,20 +612,29 @@ int policy_snapshot_new(PolicySnapshot **snapshotp,
         if (!snapshot->seclabel)
                 return error_origin(-ENOMEM);
 
+        c_rbtree_for_each_entry(node, &registry->uid_range_tree, registry_node) {
+                if (node->index.uidgid_start > uid)
+                        continue;
+                if (node->index.uidgid_end < uid)
+                        continue;
+
+                snapshot->batches[snapshot->n_batches++] = policy_batch_ref(node->batch);
+        }
+
         node = policy_registry_find_uid(registry, uid);
         if (node)
-                snapshot->batches[0] = policy_batch_ref(node->batch);
+                snapshot->batches[snapshot->n_batches++] = policy_batch_ref(node->batch);
         else
-                snapshot->batches[0] = policy_batch_ref(registry->default_batch);
-
-        ++snapshot->n_batches;
+                snapshot->batches[snapshot->n_batches++] = policy_batch_ref(registry->default_batch);
 
         while (n_gids-- > 0) {
                 node = policy_registry_find_gid(registry, gids[n_gids]);
                 if (node)
                         snapshot->batches[snapshot->n_batches++] = policy_batch_ref(node->batch);
         }
 
+        assert(snapshot->n_batches <= n_batches);
```

Comment:
Should be equal, shouldn't it?

Reply:
No, n_batches is based on n_gids, but there may not be policies for all gids.